java-squid/src/main/java/org/sonar/java/resolve/BytecodeVisitor.java

```diff
@@ -370,6 +394,39 @@ public void visitEnd() {
     }
   }
 
+
+
+  private static class TypeParameterDeclaration extends SignatureVisitor {
+
+    private final JavaSymbol symbol;
+
+    public TypeParameterDeclaration(JavaSymbol symbol) {
+      super(Opcodes.ASM5);
+      this.symbol = symbol;
+      if(symbol.isTypeSymbol()) {
+        ((JavaSymbol.TypeJavaSymbol) symbol).typeParameters = new Scope(symbol);
+      }else if (symbol.isMethodSymbol()) {
+        ((JavaSymbol.MethodJavaSymbol) symbol).typeParameters = new Scope(symbol);
+      }
+    }
+
+    @Override
+    public void visitFormalTypeParameter(String name) {
+      JavaSymbol.TypeVariableJavaSymbol typeVariableSymbol = new JavaSymbol.TypeVariableJavaSymbol(name, symbol);
+      if(symbol.isTypeSymbol()) {
+        JavaSymbol.TypeJavaSymbol typeJavaSymbol = (JavaSymbol.TypeJavaSymbol) symbol;
+        typeJavaSymbol.typeParameters.enter(typeVariableSymbol);
+        typeJavaSymbol.addTypeParameter((JavaType.TypeVariableJavaType) typeVariableSymbol.type);
+        ((JavaType.TypeVariableJavaType) typeVariableSymbol.type).bounds = Lists.newArrayList();
```

Comment:
You can avoid the repetition of this line (lines 420 & 425) by setting the bounds before entering the `if`.